src/core/api/coap_secure_api.cpp

```diff
@@ -52,98 +52,58 @@ otError otCoapSecureStart(otInstance *aInstance, uint16_t aPort)
     return instance.GetApplicationCoapSecure().Start(aPort);
 }
 
+#ifdef MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
 otError otCoapSecureSetCertificate(otInstance *   aInstance,
                                    const uint8_t *aX509Cert,
                                    uint32_t       aX509Length,
                                    const uint8_t *aPrivateKey,
                                    uint32_t       aPrivateKeyLength)
 {
-#ifdef MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
     Instance &instance = *static_cast<Instance *>(aInstance);
 
-    if (aX509Cert == NULL || aX509Length == 0 || aPrivateKey == NULL || aPrivateKeyLength == 0)
-    {
-        return OT_ERROR_INVALID_ARGS;
-    }
+    assert(aX509Cert != NULL && aX509Length != 0 && aPrivateKey != NULL && aPrivateKeyLength != 0);
 
     return instance.GetApplicationCoapSecure().SetCertificate(aX509Cert, aX509Length, aPrivateKey, aPrivateKeyLength);
-#else
-    OT_UNUSED_VARIABLE(aInstance);
-    OT_UNUSED_VARIABLE(aX509Cert);
-    OT_UNUSED_VARIABLE(aX509Length);
-    OT_UNUSED_VARIABLE(aPrivateKey);
-    OT_UNUSED_VARIABLE(aPrivateKeyLength);
-
-    return OT_ERROR_DISABLED_FEATURE;
-#endif // MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
 }
 
 otError otCoapSecureSetCaCertificateChain(otInstance *   aInstance,
                                           const uint8_t *aX509CaCertificateChain,
                                           uint32_t       aX509CaCertChainLength)
 {
-#ifdef MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
     Instance &instance = *static_cast<Instance *>(aInstance);
 
-    if (aX509CaCertificateChain == NULL || aX509CaCertChainLength == 0)
-    {
-        return OT_ERROR_INVALID_ARGS;
-    }
+    assert(aX509CaCertificateChain != NULL && aX509CaCertChainLength != 0);
 
     return instance.GetApplicationCoapSecure().SetCaCertificateChain(aX509CaCertificateChain, aX509CaCertChainLength);
-#else
-    OT_UNUSED_VARIABLE(aInstance);
-    OT_UNUSED_VARIABLE(aX509CaCertificateChain);
-    OT_UNUSED_VARIABLE(aX509CaCertChainLength);
-
-    return OT_ERROR_DISABLED_FEATURE;
-#endif // MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
 }
+#endif // MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
 
+#ifdef MBEDTLS_KEY_EXCHANGE_PSK_ENABLED
 otError otCoapSecureSetPsk(otInstance *   aInstance,
```

Comment:
Should we remove the return value `OT_ERROR_DISABLED_FEATURE` and `OT_ERROR_INVALID_ARGS ` from the function comments?